src/main/java/com/powsybl/openloadflow/ac/nr/NewtonRaphson.java

```diff
@@ -36,37 +37,33 @@
 
     private final JacobianMatrix j;
 
-    public NewtonRaphson(LfNetwork network, MatrixFactory matrixFactory, AcLoadFlowObserver observer,
-                         EquationSystem equationSystem, JacobianMatrix j, NewtonRaphsonStoppingCriteria stoppingCriteria) {
+    public NewtonRaphson(LfNetwork network, MatrixFactory matrixFactory, EquationSystem equationSystem, JacobianMatrix j,
+                         NewtonRaphsonStoppingCriteria stoppingCriteria) {
         this.network = Objects.requireNonNull(network);
         this.matrixFactory = Objects.requireNonNull(matrixFactory);
-        this.observer = Objects.requireNonNull(observer);
         this.equationSystem = Objects.requireNonNull(equationSystem);
         this.j = Objects.requireNonNull(j);
         this.stoppingCriteria = Objects.requireNonNull(stoppingCriteria);
     }
 
+    public void logLargestMismatches(double[] mismatch, EquationSystem equationSystem, int count) {
+        if (LOGGER.isTraceEnabled()) {
+            equationSystem.getSortedEquationsToSolve().keySet().stream()
+                    .map(equation -> Pair.of(equation, mismatch[equation.getColumn()]))
+                    .filter(e -> Math.abs(e.getValue()) > Math.pow(10, -7))
+                    .sorted(Comparator.comparingDouble((Map.Entry<Equation, Double> e) -> Math.abs(e.getValue())).reversed())
```

Comment:
Strange, it should be `Pair<Equation, Double>` instead of `Map.Entry<Equation, Double>`

Reply:
I don't understand, Pair has the same methods that Entry.